src/urllib3/connection.py

```diff
@@ -428,6 +441,43 @@ def connect(self):
             or self.assert_fingerprint is not None
         )
 
+    def set_tls_in_tls_required(self):
```

Comment:
Can these two methods be a variable on the `HTTPSConnection` instance instead? `self.requires_tls_in_tls`?

Reply:
Yes, good idea!